### PR TITLE
Moisture equation setup + model cleanup

### PR DESCRIPTION
**Pull request by bischtob**
**context** 
This PR allows for toggling of moisture variables in the nonhydrostatic models via dispatch on different pressure calculations. It also adds a few utility functions for other problems.

- [x] complete integration of model styles to toggle thermodynamics, moisture, etc.
- [x] add get_spaces functionality for simulations to allow discretely balanced ICs
- [x] add moisture advection equation with hyperdiffusion
- [x] add equilibrium pressure calculation

### CODE REVIEW

**Comment by bischtob** [comment]
bors try

File: src/Models/Nonhydrostatic3DModels/equations_moisture.jl [hunk 1]
```diff
@@ -9,3 +9,61 @@
...
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+
+    # auxiliary variables
+    uvw = @. Geometry.Covariant123Vector(uh) +
+       Geometry.Covariant123Vector(interp_f2c(w))
+
+    # hyperdiffusion
+    χq = @. dρq_tot = hwdiv(hgrad(ρq_tot / ρ))
```
Comment (jiahe23) [inline src/Models/Nonhydrostatic3DModels/equations_moisture.jl:58]:
```suggestion
    dρq_tot = 0.0 .* dρq_tot
    χq = @. hwdiv(hgrad(ρq_tot / ρ))
```

File: src/Models/Nonhydrostatic3DModels/equations_pressure.jl [hunk 2]
```diff
@@ -47,3 +47,41 @@ end
...
+
+    e_int = @. ρe_tot / ρ - Φ - norm(uvw)^2 / 2
+    q_tot = @. ρq_tot / ρ
+    thermo_state =
+        Thermodynamics.PhaseEquil_ρeq.(
+            params,
+            ρ,
+            e_int,
+            q_tot,
+            maxiter = 1,
```
Comment (jiahe23) [inline src/Models/Nonhydrostatic3DModels/equations_pressure.jl:81]: I am not sure about this number ...

Comment (bischtob) [reply src/Models/Nonhydrostatic3DModels/equations_pressure.jl:81]: why lol

**Pull request merged**